File: src/main/java/com/mycompany/documientos/entity/Taxonomia.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package com.mycompany.documientos.entity;

import jakarta.persistence.Basic;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.NamedQueries;
import jakarta.persistence.NamedQuery;
import jakarta.persistence.Table;
import jakarta.persistence.Temporal;
import jakarta.persistence.TemporalType;
import java.io.Serializable;
import java.util.Date;

/**
 *
 * @author daniloues
 */
@Entity
@Table(name = "taxonomia")
@NamedQueries({
    @NamedQuery(name = "Taxonomia.findAll", query = "SELECT t FROM Taxonomia t"),
    @NamedQuery(name = "Taxonomia.findByIdTaxonomia", query = "SELECT t FROM Taxonomia t WHERE t.idTaxonomia = :idTaxonomia"),
    @NamedQuery(name = "Taxonomia.findByFechaCreacion", query = "SELECT t FROM Taxonomia t WHERE t.fechaCreacion = :fechaCreacion")})
public class Taxonomia implements Serializable {

    private static final long serialVersionUID = 1L;
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Basic(optional = false)
    @Column(name = "id_taxonomia")
    private Long idTaxonomia;
    @Column(name = "fecha_creacion")
    @Temporal(TemporalType.TIMESTAMP)
    private Date fechaCreacion;
    @JoinColumn(name = "id_documento", referencedColumnName = "id_documento")
    @ManyToOne(optional = false)
    private Documento idDocumento;
    @JoinColumn(name = "id_tipo_documento", referencedColumnName = "id_tipo_documento")
    @ManyToOne(optional = false)
    private TipoDocumento idTipoDocumento;

    public Taxonomia() {
    }

    public Taxonomia(Long idTaxonomia) {
        this.idTaxonomia = idTaxonomia;
    }

    public Long getIdTaxonomia() {
        return idTaxonomia;
    }

    public void setIdTaxonomia(Long idTaxonomia) {
        this.idTaxonomia = idTaxonomia;
    }

    public Date getFechaCreacion() {
        return fechaCreacion;
    }

    public void setFechaCreacion(Date fechaCreacion) {
        this.fechaCreacion = fechaCreacion;
    }

    public Documento getIdDocumento() {
        return idDocumento;
    }

    public void setIdDocumento(Documento idDocumento) {
        this.idDocumento = idDocumento;
    }

    public TipoDocumento getIdTipoDocumento() {
        return idTipoDocumento;
    }

    public void setIdTipoDocumento(TipoDocumento idTipoDocumento) {
        this.idTipoDocumento = idTipoDocumento;
    }

    @Override
    public int hashCode() {
        int hash = 0;
        hash += (idTaxonomia != null ? idTaxonomia.hashCode() : 0);
        return hash;
    }

    @Override
    public boolean equals(Object object) {
        // TODO: Warning - this method won't work in the case the id fields are not set
        if (!(object instanceof Taxonomia)) {
            return false;
        }
        Taxonomia other = (Taxonomia) object;
        if ((this.idTaxonomia == null && other.idTaxonomia != null) || (this.idTaxonomia != null && !this.idTaxonomia.equals(other.idTaxonomia))) {
            return false;
        }
        return true;
    }

    @Override
    public String toString() {
        return "com.mycompany.documientos.entity.Taxonomia[ idTaxonomia=" + idTaxonomia + " ]";
    }
    
}

File: DocumientosFE/src/js/entidad/Taxonomia.js
export class Taxonomia {
    constructor(idTaxonomia, idDocumento, idTipoDocumento, fechaCreacion) {
        this._idTaxonomia = idTaxonomia;
        this._idDocumento = idDocumento;
        this._idTipoDocumento = idTipoDocumento;
        this._fechaCreacion = fechaCreacion;
    }

    get idTaxonomia() {
        return this._idTaxonomia;
    }

    set idTaxonomia(value) {
        this._idTaxonomia = value;
    }

    get idDocumento() {
        return this._idDocumento;
    }

    set idDocumento(value) {
        this._idDocumento = value;
    }

    get idTipoDocumento() {
        return this._idTipoDocumento;
    }

    set idTipoDocumento(value) {
        this._idTipoDocumento = value;
    }

    get fechaCreacion() {
        return this._fechaCreacion;
    }

    set fechaCreacion(value) {
        this._fechaCreacion = value;
    }

    toJSON() {
        const json = {
            idTaxonomia: this.idTaxonomia,
            idDocumento: this.idDocumento,
            idTipoDocumento: this.idTipoDocumento,
            fechaCreacion: this.fechaCreacion
        };

        return json;
    }

    static fromJSON(json) {
        return new Taxonomia(
            json.idTaxonomia,
            json.idDocumento,
            json.idTipoDocumento,
            json.fechaCreacion
        );
    }
}
